src/asm_writing/rewriter.h

```diff
@@ -578,9 +578,12 @@ class Rewriter : public ICSlotRewrite::CommitHook {
 
     void assertConsistent() {
```

Comment:
I think it's better to just add a `if (failed) return;` at the beginning of the function